go/store/nbs/store.go

```diff
@@ -1095,8 +1101,51 @@ func (nbs *NomsBlockStore) WriteTableFile(ctx context.Context, fileId string, nu
 	return err
 }
 
+// PruneTableFiles deletes old table files that are no longer referenced in the manifest.
+func (nbs *NomsBlockStore) PruneTableFiles(ctx context.Context) (err error) {
+	nbs.mu.Lock()
+	defer nbs.mu.Unlock()
+
+	nbs.mm.LockForUpdate()
+	defer func() {
+		unlockErr := nbs.mm.UnlockForUpdate()
+
+		if err == nil {
+			err = unlockErr
+		}
+	}()
+
+	for {
```

Comment:
The idea of this loop is that it's like a commit, but without a moving `root`?

Reply:
Right, we want to persist any chunks added through `nbs.Put()`, specifically via `ValueStore.WriteValue()`.
I originally had `nbs.Commit(ctx, nbs.upstream.root, nbs.upstream.root)` but that deadlocks because we already have the lock at the top of this method.